### PR TITLE
Save images on Firebase

### PR DESCRIPTION
**Pull request by betrisey**
I already implemented the saving on Firebase.
@Superdoude1999 will then update the UI to use it

### CODE REVIEW

File: app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt [hunk 1]
```diff
@@ -72,7 +76,12 @@ class ItemsViewModel @Inject constructor(
      */
     fun setItem(item: Item, callback: ((String?) -> Unit)? = null) {
         viewModelScope.launch(Dispatchers.IO) {
-            val itemId = itemRepository.set(item)
+            val uploadUrl = item.image?.let {
+                if(!it.startsWith("https://")){
+                    imageStore.store(it.toUri())
+                }
+            }
+            val itemId = itemRepository.set(item.copy(image = uploadUrl.toString()))
```
Comment (betrisey) [inline app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt:84]: You have to add an else where you return the current URL so you don't lose the existing image
```suggestion
            val uploadUrl = item.image?.let {
                if (!it.startsWith("https://")) {
                    imageStore.store(it.toUri()).toString()
                } else {
                    it
                }
            }
            val itemId = itemRepository.set(item.copy(image = uploadUrl))
```

Comment (Superdoude1999) [reply app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt]: Yup thank you for the catch, modified and implemented in the latest commit

**Review by Superdoude1999 — APPROVED** [review]
Sneaky reviews

**Review by Superdoude1999 — APPROVED** [review]
thank you sir

**Review by bafavre — APPROVED** [review]
All good, well done!

**Pull request merged**